core/src/main/scala/anorm/Row.scala

```diff
@@ -96,21 +97,22 @@ trait Row {
   def apply[B](position: Int)(implicit c: Column[B]): B =
     unsafeGet(SqlParser.get(position)(c))
 
+  @SuppressWarnings(Array("EitherGet")) // TODO: Safe alternative
   @inline def unsafeGet[T](rowparser: => RowParser[T]): T =
     (rowparser(this) match {
       case Success(v) => Right(v)
       case Error(err) => Left(err)
-    }).right.get // TODO: Safe alternative
+    }).right.get
```

Comment:
I know this isn't related to this PR, but out of interest, why does this code convert a `Try` to an `Either`, but then assume the result is a `Right` by calling `.right.get`? This means that if the original `Try` was an `Error`, we lose the original error and now get the result of `LeftProjection.get`, which always throws an `NoSuchElementException`…

Instead of all this we could just do `rowparser(this).get`…

Reply:
Good catch

Reply:
Refactored